### PR TITLE
front: separate stdcm and operationnal studies

### LINKED ISSUE #5074: front: adapt STDCM tree to refacto & isolate it from operational studies
### Description and goal

STDCM app must follow the new tree and routing ruled by the front refacto

it should be totally isolated from Operational Studies app

### Acceptance criteria

- [x] add STDCM folder in /modules, move STDCM components in it
- [x] move STDCM consts from /applications/operationalStudies/consts.ts to /applications/stdcm/consts.ts
- [x] set manageTrainSchedule i18n translation json out of operationnal studies folder

### PR DESCRIPTION
**Pull request by Yohh**
- move MODES, DEFAULT_MODE and associated types from operationnalStudies/consts.ts to main/consts.ts
- move OsrdConfState and OsrdStdcmConfState from operationnalStudies/consts.ts to reducers/osrdconf/consts.ts
- move RUNTIME_CAP from operationnalStudies/consts.ts to stdcm/consts.ts
- move allowancesConsts.ts from applications/stdcm to modules/stdcmAllowances
- update imports, add type keyword when needed

close #5074

### CODE REVIEW

**Comment by kmer2016** [comment]
Well done !

**Review by kmer2016 — APPROVED** [review]
LGTM ! Well done !

File: front/src/main/consts.ts
Comment (SharglutDev) [inline front/src/main/consts.ts:1]: There is a constants.ts file in the main utils folder. Either we should put this consts inside or move the others on this one (I vote the first option for consistency)

File: front/src/reducers/osrdconf/consts.ts
Comment (SharglutDev) [inline front/src/reducers/osrdconf/consts.ts:1]: Shouldn't the file name be types.ts ?

File: front/src/reducers/osrdconf/osrdConfCommon/__tests__/utils.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import { omit } from 'lodash';
 import { createStoreWithoutMiddleware } from 'store';
+import commonConfBuilder from 'reducers/osrdconf/osrdConfCommon/__tests__/commonConfBuilder';
```
Comment (SharglutDev) [inline front/src/reducers/osrdconf/osrdConfCommon/__tests__/utils.ts]: Better to keep all reducers imports together like you did for the other files

**Review by SharglutDev — COMMENTED** [review]
Thank you for the PR left some comments that we can discuss :)

Tested the app and stdcm still works great.

File: front/src/main/consts.ts
Comment (Yohh) [reply front/src/main/consts.ts:1]: need to talk about that at F12

File: front/src/reducers/osrdconf/consts.ts
Comment (Yohh) [reply front/src/reducers/osrdconf/consts.ts:1]: as above

File: front/src/reducers/osrdconf/osrdConfCommon/__tests__/utils.ts (hunk 1 above)
Comment (Yohh) [reply front/src/reducers/osrdconf/osrdConfCommon/__tests__/utils.ts]: done

**Review by SharglutDev — APPROVED** [review]
Lgtm

**Pull request merged**